src/Estimators/include/BipedalLocomotion/Estimators/FloatingBaseEstimator.h

```diff
@@ -0,0 +1,274 @@
+    * @param[in] encoder_speeds joint velocities measured through encoders
+    * @return True in case of success, false otherwise.
+    */
+    bool setKinematics(const Eigen::VectorXd& encoders,
+                       const Eigen::VectorXd& encoder_speeds);
+
+    /**
+    * Compute one step of the estimator
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool advance() final;
+
+    /**
+    * Get estimator outputs
+    * @return A struct containing he estimated internal states of the estiamtor and the associated covariance matrix
+    */
+    virtual const FBEOutput& get() const final;
+
+    /**
+    * Determines the validity of the object retrieved with get()
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool isValid() const final { return (m_estimator_state == State::Running); };
+
+    /**
+    * Get ModelComputations object by reference
+    * @return ModelComputations object providing information between considered model related quantities in the estimator
+    * like the base link, IMU, feet contact frames.
+    */
+    ModelComputations& modelComputations();
+
+protected:
+    /**
+    *
+    * @param[in] handler p_handler:...
+    * @return bool
+    */
+    virtual bool customInitialization(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler) { return true; };
+
+    /**
+    * Propagate the states through the prediction model, if there exists any (eg. a strap-down IMU model)
+    * @param[in] meas measurements passed as exogeneous inputs to the prediction model
+    * @param[in] dt sampling period in seconds
+    * @param[out] state previous state estimate
+    * @param[out] P previous state covariance matrix
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool predictState(const FBEMeasurements& meas,
+                              const double& dt) { return true; };
+
+    /**
+    * Update the predicted state estimates using kinematics and contact measurements
+    * @param[in] meas measurements to update the predicted states
+    * @param[in] dt sampling period in seconds
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool updateKinematics(const FBEMeasurements& meas,
+                                  const double& dt) { return true; };
+
+   /**
+   * Update the states and the associated covariance using the Kalman gain and the innovations (in an EKF based implementation)
+   * @note this function should be called from within updateKinematics in an EKF base implementation, otherwise left unused
+   * @param[in] deltaY innovation vector
+   * @param[in] H discrete-time measurement model Jacobian
+   * @param[in] R discrete-time measurement noise covariance matrix
+   * @return bool
+   */
+   virtual bool updateStates(const Eigen::VectorXd& deltaY,
+                             const Eigen::MatrixXd& H,
+                             const Eigen::MatrixXd& R) { return true; };
```

Comment:
To be honest, it is not very clear to me the meaning of the inputs. Using single letters as names may be misleading. ``H`` may mean also ``adjoint transformation``.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/79/commits/29148cbdbdb7d9e43e8b50c5434b0daeb7af4706